src/qilib/configuration_helper/instrument_adapter.py

```diff
@@ -84,3 +84,7 @@ def _filter_parameters(self, parameters: PythonJsonStructure) -> PythonJsonStruc
             PythonJsonStructure: Contains the instrument snapshot without the instrument
                                  parameters which are filtered out by this function.
         """
+
+    def close_instrument(self) -> None:
+        """ Close the wrapped QCoDeS instrument."""
+        self._instrument.close()
```

Comment:
What about the case where an instrument adapter does not create an instrument? In that case this method fails with an exception because `_instrument` is `None`.

```
from qilib.configuration_helper.adapters.common_instrument_adapter import CommonInstrumentAdapter

class TestAdapter(CommonInstrumentAdapter):
    def _filter_parameters(self, p):
        return p
    
c=TestAdapter('x')
c.close_instrument()
```
But maybe it is ok, since these adapters can overwrite the `close_instrument` if needed.

Reply:
Yes I think so, then they also need to override the `read` method implemented in the ABC base class and the `instrument` property.